packages/react/src/table/TableScroller.tsx

```diff
@@ -0,0 +1,336 @@
+			document.removeEventListener('mouseup', handleThumbRelease);
+			document.removeEventListener('touchmove', handleThumbMove);
+			document.removeEventListener('touchend', handleThumbRelease);
+		};
+	}, [handleThumbMove, handleThumbRelease, isDraggingThumb]);
+
+	const handleButtonClick = (direction: 'left' | 'right') => {
+		// Windows and keyboard left/right generally moves 40px, but it can change based on some ratio. Let's just keep it simple for now
+		const scrollAmount = direction === 'left' ? -40 : 40;
+
+		if (scrollerRef.current) {
+			scrollerRef.current.scrollLeft += scrollAmount;
+		}
+
+		if (buttonIntervalId) {
+			clearInterval(buttonIntervalId);
+			setButtonIntervalId(null);
+		}
+	};
+
+	const handleButtonPress = (
+		event: React.MouseEvent | React.TouchEvent,
+		direction: 'left' | 'right'
+	) => {
+		// Windows and keyboard left/right generally moves 40px, but it can change based on some ratio. Let's just keep it simple for now
+		const scrollAmount = direction === 'left' ? -40 : 40;
+
+		const intervalId = window.setInterval(() => {
+			if (
+				scrollerRef.current &&
+				(event.type === 'touchstart' ||
+					(event.type === 'mousedown' &&
+						'button' in event &&
+						event.button === 0))
+			) {
+				scrollerRef.current.scrollLeft += scrollAmount;
+			}
+		}, 100);
+
+		setButtonIntervalId(intervalId);
+	};
+
+	const handleButtonRelease = () => {
+		if (buttonIntervalId) {
+			clearInterval(buttonIntervalId);
+			setButtonIntervalId(null);
+		}
+	};
+
+	const handleTrackClick = (event: React.MouseEvent) => {
+		if (!scrollerRef?.current || !thumbRef?.current) {
+			return;
+		}
+
+		const thumbDimensions = thumbRef.current.getBoundingClientRect();
+
+		if (event.pageX > thumbDimensions.x + thumbDimensions.width) {
+			scrollerRef.current.scrollLeft += thumbDimensions.width * 0.95;
+		} else if (event.pageX < thumbDimensions.x) {
+			scrollerRef.current.scrollLeft -= thumbDimensions.width * 0.95;
+		}
+	};
+
+	return (
+		<Stack
+			gap={0.5}
+			css={{
+				position: 'relative',
+				width: '100%',
+			}}
+		>
+			<Box
+				css={{
+					msOverflowStyle: 'none',
+					overflowX: 'auto',
+					scrollbarWidth: 'none',
+					WebkitOverflowScrolling: 'touch',
+					width: '100%',
+					'&::-webkit-scrollbar, &::-webkit-scrollbar-thumb, &::-webkit-scrollbar-track':
+						{
+							display: 'none',
+						},
+				}}
+				focusRingFor="keyboard"
+				onScroll={repositionThumb}
+				ref={scrollerRef}
+				tabIndex={0}
+			>
+				{children}
+			</Box>
+			<Flex
+				background="body"
+				alignItems="center"
+				css={{
+					bottom: 0,
+					display: thumbWidthRatio === 1 ? 'none' : undefined,
+					left: 0,
+					position: 'sticky',
+					right: 0,
+				}}
+				flexWrap="nowrap"
+				gap={0.25}
+			>
+				<Box
+					aria-hidden
+					as="button"
+					css={{
+						appearance: 'none',
+						background: 'none',
+						cursor: 'default',
+						height: 24,
+						width: 24,
```

Comment:
Should these be in `rem`s?

Reply:
I specifically chose not to use rems since native scrollbars don't change size. It would also make the calculations slightly harder.

I could add a comment stating this rationale? Or I could change to rems?

Reply:
Yeah, I see where you're coming from. The one thing I can't help but think here is that you can actually zoom `px` sized elements so we are offering a different experience for those setting zoom levels in the browser vs those manually setting them per site with `ctrl/cmd` + `+/-`. I don't know what the best solution is here. Happy to roll with `px` but let's add a comment so that it doesn't get changed with that px-for-rem-swap ticket we have.

Reply:
Interesting. Now I'm leaning towards using rems 😅